src/lib/interfaces/high5/space/execution/index.ts

```diff
@@ -154,6 +154,8 @@ export interface StreamNode {
     inputs?: StreamNodeResolvedInputs[];
     outputs?: StreamNodeOutput[];
     additionalConnectors?: StreamNodeAdditionalConnector[];
+    color: string;
+    nodeType: NodeType;
```

Comment:
Isn't the nodeType calculated only in the engine instead of sent in the execution package?

Reply:
It gets calculated on the engine and then set into the info prop of a node result, which then gets played back to high4 execution logs. See 

backend/src/helper/ExecutionStateHelper.ts -> reportStatus -> send a High5ExecutionPatch

which is: 

```ts
export interface High5ExecutionPatch {
    high5ExecutionPatchStatus: High5ExecutionPatchStatus;
    high5ExecutionPatchLog: High5ExecutionPatchLog;
}
```

with: 
```ts
export interface High5ExecutionPatchLog {
    streamId: string;
    nodeResults: StreamSingleNodeResult[];
}
```

and finally: 
```ts
export interface StreamSingleNodeResult {
    uuid: string;
    nodeUuid: string;
    failed: boolean;
    startTimestamp: number;
    endTimestamp?: number;
    name: string;
    inputs?: StreamNodeResolvedInputs[];
    outputs?: StreamNodeOutput[];
    error?: StreamNodeResultError;
    duration?: number;
    bypassed?: boolean;
    nodeResults?: StreamSingleNodeResult[];
    info?: NodeInfo;
}
type NodeInfo = {
    runTime: number;
    catalog: {
        name?: string;
        _id?: string;
        url: string;
        version: string;
    };
};
```